pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -250,6 +356,38 @@ func (service *AlertRuleService) UpdateRuleGroup(ctx context.Context, user ident
 				New:      newRule,
 			})
 		}
+
+		// check if user has write access to all rules and can bypass the regular checks.
+		// If it cannot, check that the user is authorized to perform all the changes caused by this request
+		if can, err := service.authz.CanWriteAllRules(ctx, user); !can || err != nil {
+			if err != nil {
+				return err
+			}
```

Comment:
There are a few places like this where we can avoid checking err twice by moving around the logic.

```suggestion
		can, err := service.authz.CanWriteAllRules(ctx, user)
		if err != nil {
			return err
		}
		if !can {
```